models/User.php

```diff
@@ -354,6 +357,44 @@ public function isBanned()
         return $throttle ? $throttle->is_banned : false;
     }
 
+    //
+    // IP Recording and Throttle
+    //
+
+    /**
+     * Records the last_ip_address to reflect the last known IP for this user.
+     * @return void
```

Comment:
@daftspunk We should probably have the parameter in the docblock too:
```suggestion
     * @param string|null $ipAddress
     * @return void
```